examples/cms/src/App.tsx

```diff
@@ -0,0 +1,64 @@
+import "./App.css"
+
+import { framer, ManagedCollection, ManagedCollectionField } from "framer-plugin"
+import { useEffect, useLayoutEffect, useState } from "react"
+import { DataSource, getDataSource, getDataSourcesIds, syncCollection } from "./data"
+import { FieldMapping } from "./FieldMapping"
+import { SelectDataSource } from "./SelectDataSource"
+import { UI_DEFAULTS } from "./constants"
+
+interface AppProps {
+    collection: ManagedCollection
+    dataSourceId: string | null
+    slugFieldId: string | null
+}
+
+export function App({ collection, dataSourceId, slugFieldId }: AppProps) {
+    const [dataSource, setDataSource] = useState<DataSource | null>(null)
+    const [existingFields, setExistingFields] = useState<ManagedCollectionField[]>([])
+
+    useLayoutEffect(() => {
+        if (!dataSource) {
+            framer.showUI({
+                width: UI_DEFAULTS.SETUP_WIDTH,
+                height: UI_DEFAULTS.SETUP_HEIGHT,
+                resizable: false,
+            })
+            return
+        }
+
+        framer.showUI({
+            width: UI_DEFAULTS.MAPPING_WIDTH,
+            height: UI_DEFAULTS.MAPPING_HEIGHT,
+            minWidth: UI_DEFAULTS.MAPPING_WIDTH,
+            minHeight: UI_DEFAULTS.MAPPING_HEIGHT,
+            resizable: true,
+        })
```

Comment:
I think it would be better to make the size depends on whether a data source has been selected. Then it can render at the correct size during the initial render. So something like:

```ts
const hasDataSourceSelected = Boolean(dataSourceId || dataSource)
```

Reply:
And I would rename `dataSourceId` to something like `initialDataSourceId` or `previouslyConfiguredDataSourceId`